docs/routers/ubiquiti-usg.md

```diff
@@ -0,0 +1,66 @@
+
+```
+eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
+        inet 192.168.0.9  netmask 255.255.255.0  broadcast 192.168.0.255
+        inet6 2001:470:5:3::8ed  prefixlen 128  scopeid 0x0<global>
+        inet6 2001:470:5:3:8d70:e92e:7030:6eef  prefixlen 64  scopeid 0x0<global>
+        inet6 fe80::337f:710d:6781:4c85  prefixlen 64  scopeid 0x20<link>
+        ether dc:a6:32:0e:31:2d  txqueuelen 1000  (Ethernet)
+        RX packets 1297459  bytes 1279521260 (1.1 GiB)
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 2315988  bytes 2766370095 (2.5 GiB)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+Take note of the `inet` address, and the `inet6` address which starts with `fe` (denotes a local link or LAN address). Those are the IPs which you'll use as your DNS servers in the USG. For this example, we'll be using the `192.168.0.9` and `fe80::337f:710d:6781:4c85` IPs for the following directions
+
+### Method 1: Distribute Pi-hole as DNS server via DHCP
+
+Choosing this option will have DNS queries taking the following path:
+
+```
+(Clients) -> Pi-hole -> Upstream DNS Server
+```
+
+Go to **Settings** --> **Networks** and click on the Network line that you want to modify:
+
+![Screenshot of USG LAN](../images/usg-lan.png)
+
+Under **DHCP**, in the DHCP Server Management section click "Show options" to reveal the **DHCP DNS Server** section. Check the **Enable** box and enter your Raspi's IPv4 address here.
+
+![Screenshot of USG LAN DHCP settings](../images/usg-lan-dhcp.png)
+
+Scroll down to **IPv6**, and under the **DHCPv6/RDNSS DNS Control** section, uncheck **Auto** if it's enabled and enter Raspi's IPv6 address here. Click **Apply Changes** to have the settings persistently saved.
+
+![Screenshot of USG LAN DHCPv6 settings](../images/usg-lan-dhcpv6.png)
+
+All client devices on your network should now automatically be configured with a single IPv4 and single IPv6 address for DNS resolution in the IP stack. If you don't see your client devices update quickly enough, you *can* toggle Wi-Fi off/on or disconnect the Ethernet cable for a couple of seconds and then reconnect it.
+
+### Method 2: Using your Pi-hole as a resolve for your USG
+
+If you'd rather have your USG forward all DNS traffic to Pi-hole, you can complete this *optional* step. Doing so will have DNS queries taking the following path:
+
+```
+(Clients) -> USG -> Pi-hole -> Upstream DNS Server
+```
+
+You may want to do this for a variety of use cases. For example, you may want to have exclude a development or other test environment excluded from your normal IP stack. In my case, I wanted to enable this so I would be certain 100% of my network's DNS queries were forwarded using [DNS-over-HTTPS](https://docs.pi-hole.net/guides/dns/cloudflared/) and so I could get the benefit of individual client reporting via Pi-hole's admin interface.
```

Comment:
> I could get the benefit of individual client reporting via Pi-hole's admin interface.

How does this work if all queries are send to your USG? They will appear to Pi-hole as if they come only from this device.

Reply:
@Falc0n2k 

`(Clients) -> USG -> Pi-hole -> Upstream DNS Server`
won't show

> individual client reporting via Pi-hole's admin interface